backend/src/module/downloader/path.py

```diff
@@ -4,6 +4,7 @@
 from module.conf import settings
 from module.models import BangumiData
 
+# TODO: replace this logic with pathlib
```

Comment:
```diff
@@ -1,14 +1,10 @@
 import logging
 import re
+from pathlib import Path
 
 from module.conf import settings
 from module.models import BangumiData
 
-if ":\\" in settings.downloader.path:
-    import ntpath as path
-else:
-    import os.path as path
-
 logger = logging.getLogger(__name__)
 
 
@@ -22,7 +18,7 @@ class TorrentPath:
         subtitle_list = []
         for f in info.files:
             file_name = f.name
-            suffix = path.splitext(file_name)[-1]
+            suffix = Path(file_name).suffix
             if suffix.lower() in [".mp4", ".mkv"]:
                 media_list.append(file_name)
             elif suffix.lower() in [".ass", ".srt"]:
@@ -32,8 +28,8 @@ class TorrentPath:
     @staticmethod
     def _path_to_bangumi(save_path):
         # Split save path and download path
-        save_parts = save_path.split(path.sep)
-        download_parts = settings.downloader.path.split(path.sep)
+        save_parts = Path(save_path).parts
+        download_parts = Path(settings.downloader.path).parts
         # Get bangumi name and season
         bangumi_name = ""
         season = 1
@@ -46,7 +42,7 @@ class TorrentPath:
 
     @staticmethod
     def _file_depth(file_path):
-        return len(file_path.split(path.sep))
+        return len(Path(file_path).parts)
 
     def is_ep(self, file_path):
         return self._file_depth(file_path) <= 2
@@ -56,8 +52,8 @@ class TorrentPath:
         folder = (
             f"{data.official_title} ({data.year})" if data.year else data.official_title
         )
-        save_path = path.join(settings.downloader.path, folder, f"Season {data.season}")
-        return save_path
+        save_path = Path(settings.downloader.path) / folder / f"Season {data.season}"
+        return str(save_path)
 
     @staticmethod
     def _rule_name(data: BangumiData):
@@ -70,4 +66,4 @@ class TorrentPath:
 
     @staticmethod
     def _join_path(*args):
-        return path.join(*args)
+        return str(Path(*args))
```

Reply:
It would be better if you could define the types for method parameters, by the way 😉